File: Dataset/OOP_prepared/Python/Date_trace_4.py
from datetime import datetime

class Date:
    def __init__(self, day, month, year):
        self.day = day
        self.month = month
        self.year = year

    def get_day(self):
        return self.day

    def set_day(self, day):
        self.day = day

    def get_month(self):
        return self.month

    def set_month(self, month):
        self.month = month

    def get_year(self):
        return self.year

    def set_year(self, year):
        self.year = year

    def set_date(self, day, month, year):
        self.day = day
        self.month = month
        self.year = year

    def __str__(self):
        date_string = f"{self.year}-{self.month}-{self.day}"
        try:
            date = datetime.strptime(date_string, "%Y-%m-%d")
        except ValueError as ex:
            print(f"Error parsing date: {ex}")
            return ""
        
        new_string = date.strftime("%d/%m/%Y")
        return new_string


class TestDate:
    @staticmethod
    def main():
        d1 = Date(1, 1, 2000)
        print(d1)

if __name__ == "__main__":
    TestDate.main()

File: Dataset/OOP_prepared/Java/Date_trace_4.java
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.logging.Level;
import java.util.logging.Logger;

public class Date {
    private int day;
    private int month;
    private int year;

    public Date(int day, int month, int year) {
        this.day = day;
        this.month = month;
        this.year = year;
    }

    public int getDay() {
        return day;
    }

    public void setDay(int day) {
        this.day = day;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }
    
    public void setDate(int day, int month, int year) {
        this.day = day;
        this.month = month;
        this.year = year;
    }

    @Override
    public String toString() {
        String dateString = String.format("%d-%d-%d", year, month, day);
        java.util.Date date = null;
        
        try {
            date = new SimpleDateFormat("yyyy-MM-dd").parse(dateString);
        } catch (ParseException ex) {
            Logger.getLogger(Date.class.getName()).log(Level.SEVERE, null, ex);
        }
        
        String newstring = new SimpleDateFormat("dd/MM/yyyy").format(date);
        return  newstring;
    }
}

class TestDate {
   public static void main(String[] args) {
      Date d1 = new Date(31, 12, 1999);
      System.out.println(d1);
   }
}